content/en/docs/components/pipelines/sdk/v2/build-pipeline.md

```diff
@@ -0,0 +1,520 @@
+    means that any required import statements must be defined within the
+    function, and any helper functions must be defined within the function.
+*   The function's arguments are annotated as an [`kfp.dsl.Input`][input]
+    or [`kfp.dsl.Output`][output] artifact. These annotations let Kubeflow
+    Pipelines know to provide the path to the zipped tar file and to
+    create a path where your function stores the merged CSV file.
+*   The function is decorated with the [`kfp.dsl.component`][dsl-component]
+    annotation. You can also use this annotation to define the following:
+    
+    *   The container image that your function runs in.
+    *   Any PyPI packages that this function depends on, that are not already
+        installed on the container image.
+    *   The location to save the component specification to. You can use the
+        component specification to share this component with your colleagues.
+    
+    This annotation converts your function into a factory function that
+    creates pipeline steps. These pipeline steps execute the function you
+    defined as a part of a pipeline's workflow.
+
+Learn more about [building Python function-based components][python-function-components].
+
+The following example shows the updated `merge_csv` function.
+
+[web-download-component]: https://github.com/kubeflow/pipelines/blob/master/components/web/Download/component.yaml
+[python-function-components]: https://www.kubeflow.org/docs/components/pipelines/sdk/v2/python-function-components/
+[input]: https://github.com/kubeflow/pipelines/blob/master/sdk/python/kfp/dsl/io_types.py
+[output]: https://github.com/kubeflow/pipelines/blob/master/sdk/python/kfp/dsl/io_types.py
+[dsl-component]: https://github.com/kubeflow/pipelines/blob/master/sdk/python/kfp/dsl/_component.py
+
+
+```python
+@component(
+    base_image='python:3.7',
+    packages_to_install=['pandas==1.1.4'],
+    output_component_file='component.yaml' # This is optional. It saves the component spec for future use.
+)
+def merge_csv(tar_data: Input[Artifact], output_csv: Output[Dataset]):
+  import glob
+  import pandas as pd
+  import tarfile
+
+  tarfile.open(name=tar_data.path, mode="r|gz").extractall('data')
+  df = pd.concat(
+      [pd.read_csv(csv_file, header=None) 
+       for csv_file in glob.glob('data/*.csv')])
+  df.to_csv(output_csv.path, index=False, header=False)
+```
+
+### Build your pipeline
+
+1.  Use [`kfp.components.load_component_from_url`][load_component_from_url]
+    to load the component specification YAML for any components that you are
+    reusing in this pipeline.
+
+[load_component_from_url]: https://kubeflow-pipelines.readthedocs.io/en/latest/source/kfp.components.html?highlight=load_component_from_url#kfp.components.load_component_from_url
+
+
+```python
+web_downloader_op = kfp.components.load_component_from_url(
+    'https://raw.githubusercontent.com/kubeflow/pipelines/master/components/web/Download/component-sdk-v2.yaml')
```

Comment:
This doesn't exist yet. Is this tracked separately?

Reply:
Yes, this is in https://github.com/kubeflow/pipelines/pull/5643. Hopefully I will be able to get that merged today.

Reply:
Merged